libs/internal/include/launchdarkly/events/context_key_cache.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include <list>
+#include <string>
+#include <unordered_map>
+namespace launchdarkly::events {
+
+class ContextKeyCache {
```

Comment:
Should this be a little more generic? The cache big segments needs is mostly the same, except it is by user hash (and actually needs the expiration).

Reply:
Ah, yea might as well parameterize it now. I'll leave out the expiration until we decide.